implementations/rust/ockam/ockam_node/src/relay/processor_relay.rs

```diff
@@ -54,92 +48,36 @@ where
             Result::<()>::Ok(())
         };
 
-        #[allow(dead_code)]
-        #[derive(Debug)]
-        enum StopReason {
-            Shutdown,
-            LoopStop,
-            ProcessError(ockam_core::Error),
-            RxError(ockam_core::Error),
-        }
+        // This future resolves when a stop control signal is received
+        let shutdown_signal = async { ctrl_rx.recv().await };
 
-        #[cfg(feature = "std")]
-        let stop_reason;
+        // Then select over the two futures
         #[cfg(feature = "std")]
         tokio::select! {
```

Comment:
@antoinevg especially for this (and the same code basically in `worker_relay.rs` I would like your input

Reply:
The right thing to do here would be to implement `select!` for `no_std` so we don't need to handle the two paths differently.

I'm not proud of the hack I did to work around things in the meantime:

https://github.com/ockam-network/ockam/blob/906ebab4bf2ed18191c966e050fb178298d14f0f/implementations/rust/ockam/ockam_node/src/relay/processor_relay.rs#L66-L89